scripts/py_matter_yamltests/matter_yamltests/fixes.py

```diff
@@ -100,20 +100,31 @@ def try_add_yaml_support_for_scientific_notation_without_dot(loader):
 # accessory. But this state may not exist in the runner (as in it prevent to have multiple node ids
 # associated to a fabric...) so the 'nodeId' needs to be added back manually.
 def try_update_yaml_node_id_test_runner_state(tests, config):
+    default_identity = 'alpha'
     identities = {
-        'alpha': None if 'nodeId' not in config else config['nodeId']}
+        default_identity: None if 'nodeId' not in config else config['nodeId']}
 
     for test in tests:
         if not test.is_enabled:
             continue
 
         identity = test.identity
 
-        if test.cluster == 'CommissionerCommands':
-            if test.command == 'PairWithCode':
+        if test.cluster == 'CommissionerCommands' or test.cluster == 'DelayCommands':
```

Comment:
The reason why `TestCommissioningWindow` is now failing with this is because previously DelayCommands:WaitForCommissionee would not fall into here. It would populate the test.node_id which is what chip-repl is using.

We could move to using the test.arguments['values'] matching key value pair of nodeID.

Not sure what you think is the right approach here? Do we want test.node_id to be correct here?

Reply:
I did as you suggest since it seems like this is the `PairWithCode` behaviour in the `chip-repo` runner. 
In theory, the `DelayCommands::WaitForCommissionee` and `CommissionerCommands::PairWithCode` have little to do with the `nodeId` step value, which is used when the runner needs to send a command over the wire to a specific node for regular clusters.
I think this is OK but I am a bit confused about why you need to keep states, and notably the node id, in your runner. There may be good reason but I don't know the `chip-repl` runner implementation enough to say why.